lib/setup.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+var Hoek = require('hoek');
+
+var constructResponse = function (server) {
+  var runtime = {
+    versions: Hoek.clone(process.versions),
+    execPath: process.execPath,
+    argv: Hoek.clone(process.argv),
+    execArgv: Hoek.clone(process.execArgv),
+    cwd: process.cwd(),
+    env: Hoek.clone(process.env)
+  };
+  var connections = server.connections.map(function (connection) {
+    var plugins = {};
+
+    // TODO: Don't use _registrations and instead use a public API
+    // related issue: https://github.com/hapijs/hapi/issues/2777
+    Object.keys(connection._registrations).forEach(function (name) {
+      var registration = connection._registrations[name];
+
+      plugins[name] = {
+        name: name,
+        version: registration.version,
+        multiple: registration.multiple,
+        options: registration.options,
+        attributes: registration.register.attributes
+      };
+    });
+
+    var info = {
+      uri: connection.info.uri,
+      labels: connection.settings.labels,
+      routes: connection.table().map(function (route) {
+        return {
+          method: route.method.toUpperCase(),
+          path: route.path,
+          plugin: route.public.realm.plugin || null
+        };
+      }).sort(function (a, b) {
+        // keep nulls grouped together
+        var aPlug = a.plugin || 'z';
```

Comment:
I feel like I've commented on these same changes before. This isn't really the best way to sort, since `z` isn't the last possible string. I don't think sorting should be done by plugin. Or, if it should, it should either be done client side, or the sort criteria should be a parameter. I think it makes more sense to sort by the route, then by HTTP method.